### ISSUE TITLE
differentiate flexible and maintainable as top-level properties (was ISO-25010...)

### ISSUE THREAD

**Issue by gernotstarke** [arc42/quality.arc42.org-site#141]
https://www.iso.org/obp/ui/#iso:std:iso-iec:25010:ed-2:v1:en

**Comment by OliverThurau** [comment]
Hi Gernot,
concerning the statement "Just Eight, not 35", I notice that the "35" (or even "40" in ISO/IEC-25010:2023) are the more distinctive "second-level" qualities in the product quality model. A fair comparison would refer the 9 top-level qualities.
I did this comparison with the following result:
  
![Q42 vs IEC25010 - 1st level comparison](https://github.com/user-attachments/assets/ea1ac818-5cd8-4e19-b85c-329b72cd4a16)

-> Q42 high-level qualities are related to 7 of the ISO high-level qualities. Not addressed are compatibility and maintainability! 
One may now argue they have been proven less relevant in "my" realm. But in industry, I have to talk a lot about these two qualities and therefore wish to extend Q42's "golden eight" with two more: #compatible and #maintainable - "Just Ten" 😉

Benefit: this would increase the level of complience with IEC/ISO25019:2023.

best regards
Oliver

**Comment by gernotstarke** [comment]
Thanx Oliver.

I consider #flexible a placeholder for #maintainable, and don't particularly like the ISO-25010 distinction of these two.

"Compatible" is a valid candidate, but: what qualities would relate to this tag? I see flexibility, portability, usability, backward-compatibility as the only ones. Do you see additional ones?

**Comment by OliverThurau** [comment]
Thanks for your considerations and your point of view, Gernot,

concerning **#flexible**, my point of view is still to be able to distinguish
1.  **Flexibility** in terms of product applicability in a customer environment, i.e. qualities like adaptability, scalability, replaceability, while
2.  **Maintainability** addresses the product lifecycle processes from a developmental quality perspective, i.e. qualities like modifiability, analysability, and testability. 

I sometimes need to balance these two qualities in trade-off analysis when it comes to questions like: "we did a lot to provide a #flexible product to scale with market demands, but it still lacks of development qualities, means better traceability and thus testability." 
You can add vise flexibility points to your architecture, but that doesn't imply a highly maintainable solution. In otzher words, a #flexible product is not necessarily a #maintainable product!

With **compatibility**, I associate _(besides the 25020 mentioned qualities co-existence and interoperability)_ uses cases like
- interface stability
- product version discussions
- feature parity when it comes to e.g. re-engineering, 
- resulting in sometimes cost-intensive compatibility test efforts
- ...

**Comment by gernotstarke** [comment]
after letting this discussion sink in for month - I'm open to add maintainable as another top-level tag/label...

therefore I leave this issue open :-)

**Comment by gernotstarke** [comment]
I currently opened a new issue to re-work the domain-language: The top-level "items" (until now "properties") will be renamed "dimensions", as the term "property" is overloaded with low-level programming semantics for many developers/architects...

The lower-level qualities will become "quality characteristics" - and in lieu of that (major!) change, we can move forward with adding "maintainable" as another top-level dimension!

**Comment by OliverThurau** [comment]
I agree that **compatible** can be considered as quality attribute of the **maintainable** dimension, makes sense.
